aptos-move/framework/aptos-stdlib/sources/cryptography/bls12381.move

```diff
@@ -226,6 +243,59 @@ module aptos_std::bls12381 {
         verify_signature_share_internal(signature_share.bytes, public_key.bytes, message)
     }
 
+    #[test_only]
+    /// Generates a key pair.
+    public fun generate_keys(): (SecretKey, PublicKeyWithPoP) {
+        let (sk_bytes, pk_bytes) = generate_keys_internal();
+        let sk = SecretKey {
+            bytes: sk_bytes
+        };
+        let pkpop = PublicKeyWithPoP {
+            bytes: pk_bytes
+        };
+        (sk, pkpop)
+    }
+
+    #[test_only]
+    /// Generates a normal signature for a message with a signing key.
+    public fun sign_arbitrary_bytes(signing_key: &SecretKey, message: vector<u8>): Signature {
+        Signature {
+            bytes: sign_internal(signing_key.bytes, message)
+        }
+    }
+
+    #[test_only]
+    /// Generates a multi-signature for a message with multiple signing keys.
+    public fun multi_sign_arbitrary_bytes(signing_keys: &vector<SecretKey>, messages: vector<u8>): AggrOrMultiSignature {
+        let n = std::vector::length(signing_keys);
+        let sigs = vector[];
+        let i: u64 = 0;
+        while (i < n) {
+            let sig = sign_arbitrary_bytes(std::vector::borrow(signing_keys, i), messages);
+            std::vector::push_back(&mut sigs, sig);
+            i = i + 1;
+        };
+        let multisig = aggregate_signatures(sigs);
+        option::extract(&mut multisig)
+    }
+
+    #[test_only]
+    /// Generates an aggregated signature for n normal signatures.
+    public fun aggr_sign_arbitrary_bytes(signing_keys: &vector<SecretKey>, messages: &vector<vector<u8>>): AggrOrMultiSignature {
```

Comment:
/// Generates an aggregated signature over all mesasges in `messages`, where `signing_keys[i]` signs `messages[i]`.